python/client.py

```diff
@@ -36,7 +36,14 @@ def save_message(msg: ServiceBusReceivedMessage, download_directory):
     raw_json_s = ast.literal_eval(str(msg))
     raw_json = json.loads(raw_json_s)
     dataset = msg.subject or 'unknown'
-    file_name = f'{dataset}_{datetime.now().strftime("%y%m%dT%H%M%S_%f")}.json'
+
+    properties = msg.application_properties or {}
+    original_file_name = properties.get(b"source_file_name", "UNKNOWN").decode("utf-8")
```

Comment:
Using `"UNKNOWN"` as a fallback for if there is no value for the `b"source_file_name"` key would result in us calling `decode()` on a `str` object instead of a `bytes` variable. One way of fixing this would be to change `"UNKNOWN"` to `b"UNKNOWN"`. Alternatively we could do something a little more like this:
```
encoded_file_name = properties.get(b"source_file_name")
file_name = encoded_file_name.decode("utf-8") if encoded_file_name is not None else "UNKNOWN"
```